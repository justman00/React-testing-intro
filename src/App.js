import React from 'react';
import AnimalForm from './components/AnimalForm';

import './App.css';

function App() {
  // face un fetch

  return (
    <div className="App">
      <h1>Add New Animal</h1>
      <AnimalForm />
    </div>
  );
}

export default App;
