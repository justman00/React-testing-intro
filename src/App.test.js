import React from 'react';
import { render, screen, queryByText } from '@testing-library/react';
import App from './App';

test('renders App without errors', () => {
  render(<App />);
});

test('renders heading with the right text', () => {
  // AAA
  // arrange
  const { queryByText } = render(<App />);

  // act
  const headingText = queryByText(/add new animal/i);

  // assert
  expect(headingText).toBeInTheDocument();
});

test('renders styles accordingly', () => {
  // arrange - act
  const { container } = render(<App />);

  // assert - snapshot testing
  expect(container).toMatchSnapshot();
});
