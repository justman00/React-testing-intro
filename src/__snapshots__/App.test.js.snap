// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders styles accordingly 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Add New Animal
    </h1>
    <div
      class="animal-form"
    >
      <form>
        <label
          for="species"
        >
          Species:
        </label>
        <input
          id="species"
          name="species"
          type="text"
        />
        <label
          for="age"
        >
          Age:
        </label>
        <input
          id="age"
          name="age"
          type="text"
        />
        <label
          for="notes"
        >
          Notes:
        </label>
        <textarea
          id="notes"
          name="notes"
          type="text"
        />
        <button>
          Submit!
        </button>
      </form>
      <h4>
        Current Animals
      </h4>
      <p>
        Lion
      </p>
    </div>
  </div>
</div>
`;
